cmd/ci-operator/main.go

```diff
@@ -422,17 +423,28 @@ func (o *options) Run() error {
 			return fmt.Errorf("could not initialize namespace: %v", err)
 		}
 
+		client, err := coreclientset.NewForConfig(o.clusterConfig)
+		if err != nil {
+			return fmt.Errorf("could not get core client for cluster config: %v", err)
+		}
+		eventRecorder := eventRecorder(client, o.namespace)
+		runtimeObject := &coreapi.ObjectReference{Namespace: o.namespace}
+
+		eventRecorder.Event(runtimeObject, coreapi.EventTypeNormal, "CiJobStarted", eventJobDescription(o.jobSpec, o.namespace))
 		// execute the graph
 		suites, err := steps.Run(ctx, nodes, o.dry)
 		if err := o.writeJUnit(suites, "operator"); err != nil {
 			log.Printf("warning: Unable to write JUnit result: %v", err)
 		}
 		if err != nil {
+			eventRecorder.Event(runtimeObject, coreapi.EventTypeNormal, "CiJobFailed", eventJobDescription(o.jobSpec, o.namespace))
```

Comment:
@smarterclayton That is not working because it returns before the Event will succeed. Is there any canonical way to wait for the event recorder to send the event?
I guess time.Sleep or add a WaitGroup is not a good idea.